src/python/pants/backend/docker/util_rules/docker_build_context_test.py

```diff
@@ -251,3 +260,29 @@ def test_synthetic_dockerfile(rule_runner: RuleRunner) -> None:
             "output": {"tag": "1-1"},
         },
     )
+
+
+def test_shell_source_dependencies(rule_runner: RuleRunner) -> None:
+    rule_runner.add_to_build_file(
+        "src/docker",
+        dedent(
+            """\
+            docker_image(dependencies=[":entrypoint", ":shell"])
+            shell_source(name="entrypoint", source="entrypoint.sh")
+            shell_sources(name="shell", sources=["scripts/**/*.sh"])
+            """
+        ),
+    )
+    rule_runner.create_files("src/docker", ["Dockerfile", "entrypoint.sh"])
+    rule_runner.create_files("src/docker/scripts", ["s01.sh", "s02.sh", "random.file"])
```

Comment:
Reminder that it's preferred to use `.write_files()` instead. I'm hoping to finish removing `create_files` and friends Soon

Reply:
Right. I'll leave this for a follow up, where I go over all tests in the docker backend fixing this up.

Reply:
Ah, now that I've fixed this in another PR, this would be the only thing left to fix, so fixed it here right away.